File: src/main/java/ru/piko/pikopluginlib/EStatusPlugin.java
package ru.piko.pikopluginlib;

public enum EStatusPlugin {
    BLOCKED,
    ENABLE,
    DISABLE;

    public boolean isEnable() { return this == BLOCKED || this == ENABLE; }
    public boolean isDisable() { return this == DISABLE; }
    public boolean isBlocked() { return this == BLOCKED; }
    public boolean isUnavailable() { return this == DISABLE || this == BLOCKED; }

}

File: src/main/java/ru/piko/pikopluginlib/Api/EStatusPlugin.kt
package ru.piko.pikopluginlib.Api

enum class EStatusPlugin {
    BLOCKED_ENABLE,
    BLOCKED_DISABLE,
    ENABLE,
    DISABLE,
    UNREGISTERED;
    
    val isEnable: Boolean
        get() = this == ENABLE || this == BLOCKED_ENABLE
    val isDisable: Boolean
        get() = this == DISABLE || this == BLOCKED_DISABLE
    val isBlocked: Boolean
        get() = this == BLOCKED_ENABLE || this == BLOCKED_DISABLE
    val isUnavailable: Boolean
        get() = this == DISABLE || this == BLOCKED_ENABLE || this == BLOCKED_DISABLE || this == UNREGISTERED
    val isUnregistered: Boolean
        get() = this == UNREGISTERED
}